deploy-service/common/src/main/java/com/pinterest/deployservice/pingrequests/Ec2InstanceValidator.java

```diff
@@ -1,25 +1,44 @@
 package com.pinterest.deployservice.pingrequests;
 
-import com.pinterest.deployservice.bean.PingRequestBean;
-import com.pinterest.deployservice.common.DeployInternalException;
-import com.pinterest.deployservice.common.MetricsDataFactory;
+import java.util.List;
 
 import org.apache.commons.lang.StringUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import com.pinterest.deployservice.bean.PingRequestBean;
+import com.pinterest.deployservice.common.DeployInternalException;
+import com.pinterest.deployservice.common.MetricsDataFactory;
+
 public class Ec2InstanceValidator extends PingRequestValidator {
 
     private static final Logger LOG = LoggerFactory.getLogger(MetricsDataFactory.class);
 
     @Override
-    public void validate(PingRequestBean bean) throws Exception {
-        //Validate instance for ec2
+    public void validate(PingRequestBean bean, List<String> accountAllowList) throws Exception {
+        // Validate instance for ec2
         if (!StringUtils.startsWith(bean.getHostId(), "i-")) {
             LOG.warn("Ignore invalid id {} for host {}", bean.getHostId(), bean.getHostName());
             throw new DeployInternalException(
-                String.format("Host id %s is not a valid ec2 instance id"),
-                bean.getHostId() == null ? "null" : bean.getHostId());
+                    String.format("Host id %s is not a valid ec2 instance id"),
+                    bean.getHostId() == null ? "null" : bean.getHostId());
+        }
+
+        // Validate account id of EC2 instance
+        if (accountAllowList == null) {
+            return;
+        }
+
+        String accountId = bean.getAccountId();
+        if (accountId == null || accountId.isEmpty()) {
```

Comment:
```suggestion
        if (StringUtils.isNotBlank(accountId)) {
```

Reply:
My bad, should be `isBlank`.
Besides, you need `import org.apache.commons.lang3.StringUtils;`